services/email_service.py

```
@@ -0,0 +1,71 @@
+                except Exception as e:
+                    logger.error(f"Error processing message {message['id']}: {str(e)}")
+                    continue
+                    
+            logger.debug(f"Successfully processed {len(emails)} emails")
+            return emails
+        except Exception as e:
+            logger.error(f"Error fetching emails: {str(e)}", exc_info=True)
+            raise HTTPException(
+                status_code=500,
+                detail=f"Failed to fetch emails: {str(e)}"
+            ) 
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)